modules/apps/transfer/ibc_module.go

```diff
@@ -300,3 +305,15 @@ func (im IBCModule) OnTimeoutPacket(
 
 	return nil
 }
+
+// UnmarshalPacketData attempts to unmarshal the provided packet data bytes
+// into a FungibleTokenPacketData. This function implements the optional
+// PacketDataUnmarshaler interface required for ADR 008 support.
+func (im IBCModule) UnmarshalPacketData(bz []byte) (interface{}, error) {
```

Comment:
do we have a `PacketData` interface, and would it be helpful or have any benefit?

e.g. something like

```go
type PacketData interface {
    GetBytes() []byte
    ValidateBasic() error
}
```

Reply:
We could do this, but it wouldn't provide any functional benefits, would just clean up the API slightly. It would be an API breaking change to add which is why we have avoided it for now. I believe the interface would likely become unnecessary in the existence of a flat map packet data